File: loja-cpp-catch2/src/produto.cpp
#include "produto.hpp"

Produto::Produto(std::string nome, double preco)
    : nome_(std::move(nome)), preco_(preco) {}

const std::string& Produto::nome() const {
    return nome_;
}

double Produto::preco() const {
    return preco_;
}

File: loja-rust/src/produto.rs
#[derive(Debug, Clone)]
pub struct Produto {
    pub nome: String,
    pub preco: f64,
}

impl Produto {
    pub fn new(nome: &str, preco: f64) -> Self {
        Self {
            nome: nome.to_string(),
            preco,
        }
    }
}
